packages/firebase/index.d.ts

```diff
@@ -774,22 +774,24 @@ declare namespace firebase.firestore {
     ssl?: boolean;
 
     /**
-     * Enables the use of `Timestamp`s for timestamp fields in
-     * `DocumentSnapshot`s.
-     *
-     * Currently, Firestore returns timestamp fields as `Date` but `Date` only
-     * supports millisecond precision, which leads to truncation and causes
-     * unexpected behavior when using a timestamp from a snapshot as a part
-     * of a subsequent query.
-     *
-     * Setting `timestampsInSnapshots` to true will cause Firestore to return
-     * `Timestamp` values instead of `Date` avoiding this kind of problem. To make
-     * this work you must also change any code that uses `Date` to use `Timestamp`
-     * instead.
-     *
-     * NOTE: in the future `timestampsInSnapshots: true` will become the
-     * default and this option will be removed so you should change your code to
-     * use Timestamp now and opt-in to this new behavior as soon as you can.
+     * Specifies whether to use `Timestamp` objects for timestamp fields in
+     * `DocumentSnapshot`s. This is enabled by default and should not be
+     * disabled.
+     *
+     * Previously, Firestore returned timestamp fields as `Date` but `Date`
+     * only supports millisecond precision, which leads to truncation and
+     * causes unexpected behavior when using a timestamp from a snapshot as a
+     * part of a subsequent query.
+     *
+     * So now Firestore returns `Timestamp` values instead of `Date`, avoiding
+     * this kind of problem.
+     *
+     * To opt into the old behavior of returning `Date` objects, you can
+     * temporarily set `timestampsInSnapshots` to false.
+     *
+     * WARNING: This setting will be removed in a future release. You should
```

Comment:
Deprecation is a common way to signal this kind of "don't-use-this-legacy-thing". Does typescript have a way to formally signal that?

Even with a formal signal it's probably worth explicitly spelling the deprecation out in the text. Something like:

Maybe "This setting is deprecated and will be removed ..."

Reply:
Typescript doesn't provide a way to do it natively. You can add `@deprecated` annotation, and configure tslint to issue the warning. It's not ideal since we can't expect all our consumers to use tslint, or even typescript, so the best way might be just good documentation and a runtime warning.

Reply:
Added `@deprecated` since it doesn't hurt anything, but I will keep the log message as well, which I think is the only real way to flag deprecations in JS...